### PR TITLE
Fixes #8725 - consistency check on task data after executor start

### PR DESCRIPTION
**Pull request by iNecas**
- [x] add tests

### CODE REVIEW

**Comment by iNecas** [comment]
@jlsherrill would you mind reviewing this?

**Comment by iNecas** [comment]
Tests added

File: cc.patch [hunk 1]
```diff
@@ -0,0 +1,52 @@
+commit ce62a96f9312825cb83d6b5ff90e4fe73072af89
```
Comment (jlsherrill) [inline cc.patch]: did you mean to commit this hash?

**Comment by jlsherrill** [comment]
@iNecas i can't seem to get into the situation that i couldn't seem to get out of last week :)

Any pointers on trying to reproduce that in order to test this?

File: cc.patch (hunk 1 above)
Comment (iNecas) [inline cc.patch]: good catch :)

**Comment by iNecas** [comment]
@jlsherrill basically what's done in the tests: https://github.com/theforeman/foreman-tasks/pull/113/files#diff-27c3dbefe444dab41a4a669fe5296314R45

you can:

1.  go into the rails console; find the ForemanTasks::Task representing the currently running ListenOnCandlepin action
2. stop the dynflow executor (server in development, foreman-tasks service in production)
3. in the console, reload the task record (should be in 'stopped' state), update the state to 'running'
4. start the dynflow executor again
5. after it starts, reload the task object in console again: it should be 'stopped' again: the consistency check properly updated the state of the task

**Comment by jlsherrill** [comment]
ACK, worked well, thanks @iNecas !

**Pull request merged**